clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -4150,6 +4166,15 @@ public void reportError(String message, int userPartition, Exception e) {
       pcsList.add(partitionConsumptionStateMap.get(userPartition));
     }
     ingestionNotificationDispatcher.reportError(pcsList, message, e);
+    // Set the replica state to ERROR so that the controller can attempt to reset the partition.
+    if (!isDaVinciClient) {
+      zkHelixAdmin.get()
+          .setPartitionsToError(
```

Comment:
This is to set `EV` state as `ERROR`, right?

Reply:
I saw you mentioned that this logic would only mark `EV` state as `ERROR` when `CV` is in `ERROR` state, so how can we achieve that by the current logic?
Only the following logic today decides whether to propagate `ERROR` to `CV/EV` IIUC.

Reply:
Shall we add a config to control this behavior?